bytesource/bytesource.go

```diff
@@ -0,0 +1,61 @@
+package bytesource
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math/rand"
+)
+
+// ByteSource implements rand.Source64 determined by a slice of bytes. The random numbers are
+// generated from each 8 bytes in the slice, until the last bytes are consumed, from which a
+// fallback pseudo random source is created in case more random numbers are required.
+// It also exposes a `bytes.Reader` API, which lets consume the bytes directly.
+type ByteSource struct {
+	*bytes.Reader
+	fallback rand.Source
+}
+
+// New returns a new ByteSource from a given slice of bytes.
+func New(input []byte) *ByteSource {
+	if len(input) == 0 {
+		panic("ByteSource was initiated with empty input")
```

Comment:
hm, mention this in the function comment, or (my preference) use a seed of 0 in this case?